src/main/kotlin/com/hibob/academy/types/StoreService.kt

```diff
@@ -0,0 +1,69 @@
+package com.hibob.academy.types
+
+import com.hibob.types.*
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+        return cart.associate {
+            it.clientId to checkout(it, payment)
+        }
+    }
+}
+
+fun fail(message: String): Nothing {
```

Comment:
I would move all the function to  StoreService and make it private.
don't want to create a global function with specifice implemantion,